Dfe.Academies.External.Web/Services/ConversionApplicationRetrievalService.cs

```diff
@@ -337,47 +351,17 @@ public Status CalculateTrustStatus(ConversionApplication? conversionApplication)
 	///<inheritdoc/>
 	public Status CalculateJoinAMatTrustStatus(ConversionApplication? conversionApplication)
 	{
-		// need 3 bools to represent each sub-section. completed = yes/no
-		// 1) applicationselecttrust :- !string.IsNullOrWhiteSpace(conversionApplication.JoinTrustDetails?.TrustName) = complete
-		// 2) applicationschooltrustconsent - 3 steps :-
-		// a) step 1 (ApplicationSchoolTrustConsent) = conversionApplication.JoinTrustDetails != null ??? as this step is docs
-		// b) step 2 (ApplicationSchoolChangesToATrust) = conversionApplication.JoinTrustDetails.ChangesToTrust.HasValue = complete
-		// c) step 3 (ApplicationSchoolLocalGovernanceArrangements) = conversionApplication.JoinTrustDetails.ChangesToLaGovernanceExplained.HasValue
-
-		if (conversionApplication != null && conversionApplication.JoinTrustDetails != null)
-		{
-			BitArray statuses = new BitArray(3);
-			statuses.Set(0, !string.IsNullOrWhiteSpace(conversionApplication.JoinTrustDetails.TrustName)); // ONLY set to false IF EMPTY!
-			statuses.Set(1, conversionApplication.JoinTrustDetails!.ChangesToTrust.HasValue);
-			statuses.Set(2, conversionApplication.JoinTrustDetails!.ChangesToLaGovernance.HasValue);
-
-			////bool hasAnyFalse = statuses.Cast<bool>().Contains(false);
-			////bool hasAnyTrue = statuses.Cast<bool>().Contains(true);
-
-			int falseCount = (from bool m in statuses
-				where !m
-				select m).Count();
-
-			//// need to do a count of false. if falseCount = 3 - status = NotStarted
-			//// need to do a count of false. if falseCount = >=1 || falseCount <=2 - status = InProgress
-			//// need to do a count of false. if falseCount = 0 - status = Completed
-			if (falseCount == 3)
-			{
-				return Status.NotStarted;
-			}
-			else if (falseCount > 0)
-			{
-				return Status.InProgress;
-			}
-			else
-			{
-				return Status.Completed;
-			}
-		}
-		else
-		{
-			return Status.NotStarted;
-		}
+		if (!string.IsNullOrWhiteSpace(conversionApplication?.JoinTrustDetails?.TrustName)
+		    && conversionApplication.JoinTrustDetails.ChangesToTrust.HasValue
```

Comment:
don't think there's any reason to rewrite this?

Reply:
I just found it a bit too complicated to read is all, I thought it could be made a bit more streamlined with just some logical operators